includes/class-wc-payments.php

```diff
@@ -364,32 +366,40 @@ public static function init() {
 		self::$platform_checkout_util              = new Platform_Checkout_Utilities();
 
 		self::$legacy_card_gateway = new CC_Payment_Gateway( self::$api_client, self::$account, self::$customer_service, self::$token_service, self::$action_scheduler_service, self::$failed_transaction_rate_limiter, self::$order_service );
-		$payments_checkout         = new WC_Payments_Checkout( self::$legacy_card_gateway, self::$platform_checkout_util, self::$account, self::$customer_service );
-
-		if ( WC_Payments_Features::is_upe_enabled() ) {
-			$payment_method_classes = [
-				CC_Payment_Method::class,
-				Bancontact_Payment_Method::class,
-				Sepa_Payment_Method::class,
-				Giropay_Payment_Method::class,
-				Sofort_Payment_Method::class,
-				P24_Payment_Method::class,
-				Ideal_Payment_Method::class,
-				Becs_Payment_Method::class,
-				Eps_Payment_Method::class,
-				Link_Payment_Method::class,
-			];
+
+		$payment_methods        = [];
+		$payment_method_classes = [
+			CC_Payment_Method::class,
+			Bancontact_Payment_Method::class,
+			Sepa_Payment_Method::class,
+			Giropay_Payment_Method::class,
+			Sofort_Payment_Method::class,
+			P24_Payment_Method::class,
+			Ideal_Payment_Method::class,
+			Becs_Payment_Method::class,
+			Eps_Payment_Method::class,
+			Link_Payment_Method::class,
+		];
+		foreach ( $payment_method_classes as $payment_method_class ) {
+			$payment_method                               = new $payment_method_class( self::$token_service );
+			$payment_methods[ $payment_method->get_id() ] = $payment_method;
+		}
+		if ( WC_Payments_Features::is_upe_split_enabled() ) {
 			foreach ( $payment_method_classes as $class ) {
 				$payment_method = new $class( self::$token_service );
```

Comment:
The `UPE_Payment_Gateway` requires an array of payment methods and consequently so does the `UPE_Split_Payment_Gateway`, since it triggers the parent class's constructor. I've made the looping more efficient at the cost of some unavoidable code duplication in 3c3d75d46cc77cbec083a743953ab813f10912b8. Let me know if this will suffice. 🙏